### ISSUE TITLE
Implement Semantic Caption Matching for Florence Validation

### ISSUE THREAD

**Issue by mikesvoboda** [mikesvoboda/nemotron-v3-home-security-intelligence#4500]
## Problem Statement

The comparison engine uses exact string matching for Florence caption validation, causing false failures when semantically equivalent terms are used.

## Example Failure

**Expected Keywords:** `['person', 'package']`

**Actual Caption:**

> "The image shows a man wearing a cap and gloves, holding a **cardboard box** in his hands, standing on a doormat in front of a house..."

**Result:** FAIL

**Reason:** "cardboard box" ≠ "package" (semantically equivalent but string mismatch)

## Impact

* 10 out of 12 tested scenarios failed `florence_caption.must_contain` check
* Creates noise in test results
* Obscures real caption quality issues
* Makes load testing unreliable

## Current Implementation

**Location:** `scripts/synthetic/comparison_engine.py`

```python
# Current: exact string matching
def check_caption_keywords(caption: str, keywords: list[str]) -> bool:
    caption_lower = caption.lower()
    for keyword in keywords:
        if keyword.lower() not in caption_lower:
            return False
    return True
```

## Acceptance Criteria

1. \[ \] Semantic matching recognizes equivalent terms
2. \[ \] Configurable similarity threshold
3. \[ \] Support for synonym groups
4. \[ \] Backward compatible with existing expected_labels.json

### Option A: Synonym Expansion (Simple)

```python
SYNONYMS = {
    "package": {"package", "box", "parcel", "delivery", "cardboard box"},
    "person": {"person", "man", "woman", "individual", "figure", "someone"},
    "car": {"car", "vehicle", "automobile", "sedan", "suv"},
    "door": {"door", "entrance", "entryway", "doorway"},
    "dog": {"dog", "canine", "puppy", "pet"},
    "cat": {"cat", "feline", "kitten"},
    # ... more synonyms
}

def check_caption_keywords_with_synonyms(caption: str, keywords: list[str]) -> bool:
    caption_lower = caption.lower()
    for keyword in keywords:
        keyword_lower = keyword.lower()
        synonyms = SYNONYMS.get(keyword_lower, {keyword_lower})
        if not any(syn in caption_lower for syn in synonyms):
            return False
    return True
```

### Option B: CLIP Embedding Similarity (More Robust)

```python
import torch
from transformers import CLIPModel, CLIPTokenizer

def check_caption_semantic(caption: str, keywords: list[str], threshold: float = 0.7) -> bool:
    model = CLIPModel.from_pretrained("openai/clip-vit-base-patch32")
    tokenizer = CLIPTokenizer.from_pretrained("openai/clip-vit-base-patch32")
    
    caption_embedding = encode_text(caption)
    
    for keyword in keywords:
        keyword_embedding = encode_text(keyword)
        similarity = cosine_similarity(caption_embedding, keyword_embedding)
        if similarity < threshold:
            return False
    return True
```

### Option C: Use Existing CLIP Service (Recommended)

The system already has a CLIP service at `localhost:8093`. Use it for semantic matching:

```python
async def check_caption_semantic_via_clip(
    caption: str, 
    keywords: list[str],
    clip_url: str = "http://localhost:8093",
    threshold: float = 0.6
) -> tuple[bool, dict[str, float]]:
    """Check if caption semantically contains keywords using CLIP embeddings."""
    
    async with httpx.AsyncClient() as client:
        # Get caption embedding
        caption_resp = await client.post(
            f"{clip_url}/embed/text",
            json={"text": caption}
        )
        caption_emb = caption_resp.json()["embedding"]
        
        results = {}
        for keyword in keywords:
            # Get keyword embedding
            kw_resp = await client.post(
                f"{clip_url}/embed/text",
                json={"text": keyword}
            )
            kw_emb = kw_resp.json()["embedding"]
            
            # Compute cosine similarity
            similarity = cosine_similarity(caption_emb, kw_emb)
            results[keyword] = similarity
        
        all_pass = all(sim >= threshold for sim in results.values())
        return all_pass, results
```

## Expected Labels Format Update

Support both exact and semantic matching:

```json
{
  "florence_caption": {
    "must_contain": ["person", "package"],
    "must_contain_semantic": ["delivery activity", "residential setting"],
    "must_not_contain": ["weapon", "violence"],
    "semantic_threshold": 0.65
  }
}
```

## Files to Modify

* `scripts/synthetic/comparison_engine.py` - Add semantic matching
* `scripts/load_test_ai_pipeline.py` - Use semantic matching for caption validation
* `data/synthetic/*/expected_labels.json` - Optionally add semantic keywords

## Testing

```bash
# Test semantic matching
python -c "
from scripts.synthetic.comparison_engine import check_caption_semantic

caption = 'a man holding a cardboard box'
keywords = ['person', 'package']
result = check_caption_semantic(caption, keywords)
print(f'Match: {result}')  # Should be True
"

# Run load test with semantic matching
uv run scripts/load_test_ai_pipeline.py --quick --verbose
```

## Performance Considerations

* Option A (synonyms): Fast, no API calls, but limited vocabulary
* Option B (local CLIP): Slower startup, \~50ms per comparison
* Option C (CLIP service): \~10ms per comparison (service already warm)
